lib/task_005.rb

```diff
@@ -0,0 +1,124 @@
+# 0,0 -> 8,8
+# 5,5 -> 8,2
+#
+# Each line of vents is given as a line segment in the format x1,y1 -> x2,y2 where x1,y1 are the coordinates of
+# one end the line segment and x2,y2 are the coordinates of the other end. These line segments include the points
+# at both ends. In other words:
+#
+#     An entry like 1,1 -> 1,3 covers points 1,1, 1,2, and 1,3.
+#     An entry like 9,7 -> 7,7 covers points 9,7, 8,7, and 7,7.
+#
+# For now, only consider horizontal and vertical lines: lines where either x1 = x2 or y1 = y2.
+#
+# So, the horizontal and vertical lines from the above list would produce the following diagram:
+#
+# .......1..
+# ..1....1..
+# ..1....1..
+# .......1..
+# .112111211
+# ..........
+# ..........
+# ..........
+# ..........
+# 222111....
+#
+# In this diagram, the top left corner is 0,0 and the bottom right corner is 9,9. Each position is shown as the
+# number of lines which cover that point or . if no line covers that point. The top-left pair of 1s, for example,
+# comes from 2,2 -> 2,1; the very bottom row is formed by the overlapping lines 0,9 -> 5,9 and 0,9 -> 2,9.
+#
+# To avoid the most dangerous areas, you need to determine the number of points where at least two lines overlap.
+#  In the above example, this is anywhere in the diagram with a 2 or larger - a total of 5 points.
+#
+# Consider only horizontal and vertical lines. At how many points do at least two lines overlap?
+
+def parse_input(input_data)
+  input_data.split("\n").map { |line| parse_line(line) }
+end
+
+def parse_line(line)
+  point1, point2 = line.split(" -> ")
+
+  {
+    point1: parse_point(point1),
+    point2: parse_point(point2)
+  }
+end
+
+def parse_point(point)
+  point.split(',').map { |value| value.to_i }
+end
+
+def create_diagram(lines)
+  x = lines.max_by { |line| line[:point1][0] || line[:point2][0] }
+  y = lines.max_by { |line| line[:point1][1] || line[:point2][1] }
+
+  size_x = x[:point1][0] > x[:point2][0] ? x[:point1][0] + 1 : x[:point2][0] + 1
+  size_y = y[:point1][1] > y[:point2][1] ? y[:point1][1] + 1 : y[:point2][1] + 1
+  diagram = []
+  size_y.times do
+    diagram << Array.new(size_x, 0)
+  end
+  diagram
+end
+
+def mark_lines(lines, diagram)
+  lines.each { |line| mark_line(line, diagram) }
+  diagram
+end
+
+def mark_line(line, diagram)
+  if line[:point1][0] == line[:point2][0]
+    range = line_range(line[:point1][1], line[:point2][1])
+    while range[0] <= range[1]
+      diagram[range[0]][line[:point1][0]] += 1
+      range[0] += 1
+    end
+  end
+  if line[:point1][1] == line[:point2][1]
+    range = line_range(line[:point1][0], line[:point2][0])
+    while range[0] <= range[1]
+      diagram[line[:point1][1]][range[0]] += 1
+      range[0] += 1
+    end
+  end
+  diagram
+end
+
+def line_range(value1, value2)
+  value1 < value2 ? [value1, value2] : [value2, value1]
+end
+
+def calculate_answer(diagram)
+  i = 0
+  diagram.each do |ary|
+    i += ary.count { |value| value >= 2 }#? i += 1 : nil }
+  end
+  i
```

Comment:
```suggestion
  diagram.sum do |ary|
    ary.count { |value| value >= 2 }#? i += 1 : nil }
  end
```

Reply:
And I would try to use more meaningful names:

```suggestion
  diagram.sum do |row|
    row.count { |coverage| coverage >= 2 }
  end
```

Reply:
Yes, looks much better.